src/native/corehost/browserhost/libBrowserHost.js

```diff
@@ -0,0 +1,536 @@
+        Object.assign(assert, assertLocal);
+        Object.assign(jsEngine, jsEngineLocal);
+    }
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    function runtimeExportsToTable(map) {
+        return [];
+    }
+    function runtimeExportsFromTable(table, runtime) {
+        Object.assign(runtime, {});
+    }
+    function hostNativeExportsToTable(map) {
+        return [
+            map.registerDllBytes,
+            map.isSharedArrayBuffer,
+        ];
+    }
+    function hostNativeExportsFromTable(table, native) {
+        const nativeLocal = {
+            registerDllBytes: table[0],
+            isSharedArrayBuffer: table[1],
+        };
+        Object.assign(native, nativeLocal);
+    }
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    function interopJavaScriptNativeExportsToTable(map) {
+        return [];
+    }
+    function interopJavaScriptNativeExportsFromTable(table, interop) {
+        const interopLocal = {};
+        Object.assign(interop, interopLocal);
+    }
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    function nativeBrowserExportsToTable(map) {
+        return [];
+    }
+    function nativeBrowserExportsFromTable(table, interop) {
+        const interopLocal = {};
+        Object.assign(interop, interopLocal);
+    }
+
+    // Licensed to the .NET Foundation under one or more agreements.
+    // The .NET Foundation licenses this file to you under the MIT license.
+
+    // Licensed to the .NET Foundation under one or more agreements.
+    // The .NET Foundation licenses this file to you under the MIT license.
+    // see also `reserved` in `rollup.config.defines.js`
+
+    // Licensed to the .NET Foundation under one or more agreements.
+    // The .NET Foundation licenses this file to you under the MIT license.
+    const loadedAssemblies = {};
+    function registerDllBytes(bytes, asset) {
+        const sp = Module.stackSave();
+        try {
+            const sizeOfPtr = 4;
+            const ptrPtr = Module.stackAlloc(sizeOfPtr);
+            if (Module._posix_memalign(ptrPtr, 16, bytes.length)) {
+                throw new Error("posix_memalign failed");
+            }
+            const ptr = Module.HEAPU32[ptrPtr >> 2];
+            Module.HEAPU8.set(bytes, ptr);
+            loadedAssemblies[asset.name] = { ptr, length: bytes.length };
+        }
+        finally {
+            Module.stackRestore(sp);
+        }
+    }
+    // bool browserHostExternalAssemblyProbe(const char* pathPtr, /*out*/ void **outDataStartPtr, /*out*/ int64_t* outSize);
+    function browserHostExternalAssemblyProbe(pathPtr, outDataStartPtr, outSize) {
+        const path = Module.UTF8ToString(pathPtr);
+        const assembly = loadedAssemblies[path];
+        if (!assembly) {
+            return false;
+        }
+        Module.HEAPU32[outDataStartPtr >> 2] = assembly.ptr;
+        // upper bits are cleared by the C caller
```

Comment:
I don't understand this comment, but it doesn't seem important

Reply:
simplfied